runtime/vm/ValueTypeHelpers.cpp

```diff
@@ -146,10 +146,13 @@ findJ9ClassInFlattenedClassCache(J9FlattenedClassCache *flattenedClassCache, U_8
         J9Class *clazz = NULL;
 
         for (UDATA i = 0; i < length; i++) {
-                J9UTF8* currentClassName = J9ROMCLASS_CLASSNAME(J9_VM_FCC_ENTRY_FROM_FCC(flattenedClassCache, i)->clazz->romClass);
-                if (J9UTF8_DATA_EQUALS(J9UTF8_DATA(currentClassName), J9UTF8_LENGTH(currentClassName), className, classNameLength)) {
-                        clazz = J9_VM_FCC_CLASS_FROM_ENTRY(J9_VM_FCC_ENTRY_FROM_FCC(flattenedClassCache, i));
-                        break;
+                J9FlattenedClassCacheEntry *entry = J9_VM_FCC_ENTRY_FROM_FCC(flattenedClassCache, i);
+                if (!J9_VM_FCC_ENTRY_IS_STATIC_FIELD(entry)) {
+                        J9UTF8* currentClassName = J9ROMCLASS_CLASSNAME(entry->clazz->romClass);
```

Comment:
There is an existing macro `J9_VM_FCC_CLASS_FROM_ENTRY()` to get the `J9Class`. We can then use the `J9Class` to get the romClass and className.

Reply:
updated in [e9539c9](https://github.com/eclipse-openj9/openj9/commit/e9539c94daf4337bc0c005707c10b29812b7ed33)